Autocomplete attribute has valid value- editorial changes in response to TF feedback (#1534)
act-rules/act-rules.github.io: _rules/autocomplete-valid-value-73f2c2.md

This rule applies to any HTML `input`, `select` and `textarea` element with an `autocomplete` [attribute value][] that is neither empty (`""`) nor only [ASCII whitespace][], except if one of the following is true:

- **hidden**: the element is not [visible][], and not [included in the accessibility tree][]; or

> - **disabled**: the test target is the [disabled element]; or

Reviewer: ```suggestion
- **disabled**: the element is a [disabled element]; or
```
all conditions starts with "the element" rather than "the test target".

> - **disabled**: the element has the `disabled` attribute or the `aria-disabled` [attribute value][] of `true`; or

Reviewer: This should probably use https://act-rules.github.io/glossary/#disabled-element instead.

Also, this question the need for the last assumption. I essentially understood the rule (+assumption) as looking only at `aria-disabled` because it assumed that if the element is disabled (common meaning), this is reflected in the `aria-disabled` attribute, notably, if `disabled` is set, then `aria-disabled` is also set (by default) (see https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings / disabled, and actually if trying to set both, then `disabled` seems to take precedence).

I think that the definition (and its assumption) is capturing this idea even better (it was written after this rule) and should be used here also.

Author: The disabled element definition, as far as I understand it, applies to elements such as `<button disabled>Submit</button>` or to elements that are in the shadow DOM. It does not seem to apply to elements such as `<button aria-disabled>Submit</button>`. I'm trying to cover both scenarios but I agree that we should use the definition of the disabled element instead of "disabled" attribute. 

Am I right in saying that you're referring to `<button disabled aria-disabled>Submit</button>`?

Author: Sorry, I think that the penny dropped, we should use the disabled element definition because `disabled` is mapped to platform-specific accessibility API's as `aria-disabled="true"`, is that correct?

Reviewer: The second point in the definition is about `aria-disabled` (and should read "inclusive ancestor", rather than "ancestor" 🙈 )

"shadow including" ancestor means that it indeed apply to elements in the shadow DOM, but not only. `<div><span></span></div>`, the `div` is both an ancestor and a shadow including ancestor (i.e., it's shadow including, not light excluding).

Copying DOM standard, emphasise mine:
> An object A is a shadow-including descendant of an object B, if **A is a descendant of B**, or A’s root is a shadow root and A’s root’s host is a shadow-including inclusive descendant of B.
>
> A shadow-including inclusive descendant is an object or one of its shadow-including descendants.
>
> An object A is a shadow-including ancestor of an object B, if and only if B is a shadow-including descendant of A.
>
> A shadow-including inclusive ancestor is an object or one of its shadow-including ancestors.

So, regular (inclusive) ancestors/descendants are also shadow-including ones.

Author: Took a while, but got it in the end. Should be fixed now. Thanks!

```html
<form autocomplete="off">
	<label>Birthday month

>         <select id="bday" autocomplete="bday-month">

Reviewer: ```suggestion
        <select autocomplete="bday-month">
```
No need for an `id` with an implicit label.

```html
<form autocomplete="off">

> 	<label for="bday">Birthday month</label>

Reviewer: I think we should stick to implicit labels (`<label>Foo <widget>…</widget> </label>`) rather than explicit ones (`<label for="…">`) because this is not the core of the rule (this is not a rule about labels, so no need to show all ways to give a label), all the existing cases are using implicit labels, and it limits errors in IDs.

This `autocomplete` [attribute value][] only has the required token, and is valid for a `textarea` element. Mixing upper and lower case letters is allowed for `autocomplete` attributes.

```html

> <label for="sAddress">Street address</label> <textarea id="sAdress" autocomplete="Street-Address"></textarea>

Reviewer: ```suggestion
<label for="sAddress">Street address</label> <textarea id="sAddress" autocomplete="Street-Address"></textarea>
```
(or rather use implicit label)

This `autocomplete` [attribute value][] only has the required token, and is valid for a `textarea` element. Mixing upper and lower case letters is allowed for `autocomplete` attributes.

```html

> <label> Street address<textarea id="sAddress" autocomplete="Street-Address"></textarea></label> 

Reviewer: ```suggestion
<label> Street address<textarea autocomplete="Street-Address"></textarea></label> 
```
